Mathlib/Data/UnionFind.lean

```diff
@@ -217,7 +217,7 @@ def findAux (self : UnionFind α) (x : Fin self.size) :
       m.Models self.arr ∧ m'.Models s ∧ m'.rank = m.rank ∧
       (∃ hr, (m'.parent ⟨root, hr⟩).1 = root) ∧
       m.rank x ≤ m.rank root := by
-  let y := self.arr[x].parent
+  let y := (self.arr[x]'(Fin.isLt x)).parent
```

Comment:
What's this notation even?

Reply:
Array indexing with explicit proof that the index is in range. I didn't check why this worked before.